fusor-ember-cli/app/routes/assign-nodes.js

```diff
@@ -5,32 +5,77 @@ import DeploymentRouteMixin from "../mixins/deployment-route-mixin";
 export default Ember.Route.extend(DeploymentRouteMixin, {
 
   model() {
-      var deploymentId = this.modelFor('deployment').get('id');
-      return Ember.RSVP.hash({
-          plan: this.store.findRecord('deployment-plan', deploymentId),
-          images: this.store.query('image', {deployment_id: deploymentId}),
-          nodes: this.store.query('node', {deployment_id: deploymentId}),
-          profiles: this.store.query('flavor', {deployment_id: deploymentId}),
-      });
+    var deploymentId = this.modelFor('deployment').get('id');
+    return Ember.RSVP.hash({
+      plan: this.store.findRecord('deployment-plan', deploymentId),
+      images: this.store.query('image', {deployment_id: deploymentId}),
+      nodes: this.store.query('node', {deployment_id: deploymentId}),
+      profiles: this.store.query('flavor', {deployment_id: deploymentId}),
+    });
   },
 
   setupController(controller, model) {
     controller.set('model', model);
+    this.updateRoles();
     this.fixBadDefaults();
   },
 
+  actions: {
+    updateRoles() {
+      this.updateRoles();
```

Comment:
I wanted to reuse this code from both the controller and route to avoid code duplication.  Unfortunately, calling send from the route resulted in an unhandled action, so I broke it out into a method and called it from both setupController and the action.